operator/pkg/apis/operator/v1alpha1/type.go

```diff
@@ -0,0 +1,771 @@
+	// control plane on.
+	// This can be a hostname, hostname:port, IP or IP:port.
+	// +optional
+	APIEndpoint string `json:"apiEndpoint,omitempty"`
+
+	// SecretRef represents the secret contains mandatory credentials to
+	// access the cluster.
+	// The secret should hold credentials as follows:
+	// - secret.data.token
+	// - secret.data.caBundle
+	// +optional
+	SecretRef *LocalSecretReference `json:"secretRef,omitempty"`
+
+	// Networking holds configuration for the networking topology of the cluster.
+	// +optional
+	Networking *Networking `json:"networking,omitempty"`
+}
+
+// ConditionType declarative karmada condition type of karmada installtion.
+type ConditionType string
+
+const (
+	// Unknown represent a condition type the karmada not be reconciled by operator
+	// or unpredictable condition.
+	Unknown ConditionType = "Unknown"
+
+	// Ready represent a condition type the all installtion process to karmada have compaleted.
+	Ready ConditionType = "Ready"
+)
+
+// KarmadaStatus difine the most recently observed status of the Karmada.
+type KarmadaStatus struct {
+	// ObservedGeneration is the last observed generation.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// after the karmada installed, restore the kubeconfig to secret.
+	// +optional
+	SecretRef *LocalSecretReference `json:"secretRef,omitempty"`
+
+	// KarmadaVersion represente the karmada version.
+	// +optional
+	KarmadaVersion string `json:"karmadaVersion,omitempty"`
+
+	// KubernetesVersion represente the karmada-apiserver version.
+	// +optional
+	KubernetesVersion string `json:"kubernetesVersion,omitempty"`
+
+	// Conditions represents the latest available observations of a karmada's current state.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// LocalSecretReference is a reference to a secret within the enclosing
+// namespace.
+type LocalSecretReference struct {
+	// Namespace is the namespace for the resource being referenced.
+	Namespace string `json:"namespace,omitempty"`
+
+	// Name is the name of resource being referenced.
+	Name string `json:"name,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KarmadaList is a list of Karmadas.
+type KarmadaList struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []Karmada `json:"items"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:path=karmadaonborad,scope=Namespaced,categories={karmada-io}
+// +kubebuilder:printcolumn:JSONPath=`.spec.syncMode`,name="Mode",type=string
+// +kubebuilder:printcolumn:JSONPath=`.metadata.creationTimestamp`,name="Age",type=date
+
+// KarmadaOnborad enables declarative join member cluster to karmada.
+type KarmadaOnborad struct {
```

Comment:
Did we ever talk about this API?

Reply:
I know this API is added by https://github.com/karmada-io/karmada-operator/pull/4. 
I think we all missed it from reviewing.

For this PR, I suppose we can drop it and we can add it back when we needed it.

Reply:
This API we haven't talked about yet. 
ok, I remove it first